### PR TITLE
Make printf great again

### PR DESCRIPTION
**Pull request by DrZlo13**
# What's new

- Functions for stdout are now implemented on our side. 
- Added ability to get FuriThread instance of current thread.
- Stdglue moved to thread.
- Size of the code and the occupied RAM has decreased.

# Verification 

- Check that the output to uart and cli has not changed.
- Run unit tests.

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/desktop/views/desktop_view_debug.c [hunk 1]
```diff
@@ -98,7 +98,9 @@ void desktop_debug_render(Canvas* canvas, void* model) {
             (remaining == (uint32_t)(-1) ? remaining : 0));
         canvas_draw_str(canvas, 5, 29 + STATUS_BAR_Y_SHIFT, buffer);
 
-        snprintf(buffer, 64, "%s", asctime(localtime((const time_t*)&m->timestamp)));
+        // even if timestamp is uint64_t, it's safe to cast it to uint32_t, because furi_hal_rtc_datetime_to_timestamp only returns uint32_t
+        snprintf(buffer, 64, "%ld", (uint32_t)m->timestamp);
```
Comment (hedger) [inline applications/desktop/views/desktop_view_debug.c]: #define or sizeof pls

File: applications/rpc/rpc_storage.c [hunk 2]
```diff
@@ -541,7 +541,7 @@ static void rpc_system_storage_md5sum_process(const PB_Main* request, void* cont
         (void)md5sum_size;
         furi_assert(hash_size <= ((md5sum_size - 1) / 2));
         for(uint8_t i = 0; i < hash_size; i++) {
-            md5sum += sprintf(md5sum, "%02x", hash[i]);
+            md5sum += snprintf(md5sum, 33, "%02x", hash[i]);
```
Comment (hedger) [inline applications/rpc/rpc_storage.c]: use md5sum_size

File: applications/subghz/scenes/subghz_scene_receiver_config.c [hunk 3]
```diff
@@ -73,8 +73,9 @@ static void subghz_scene_receiver_config_set_frequency(VariableItem* item) {
 
     if(subghz->txrx->hopper_state == SubGhzHopperStateOFF) {
         char text_buf[10] = {0};
-        sprintf(
+        snprintf(
             text_buf,
+            10,
```
Comment (hedger) [inline applications/subghz/scenes/subghz_scene_receiver_config.c]: sizeof()

File: applications/subghz/scenes/subghz_scene_receiver_config.c [hunk 4]
```diff
@@ -106,8 +107,9 @@ static void subghz_scene_receiver_config_set_hopping_runing(VariableItem* item)
     variable_item_set_current_value_text(item, hopping_text[index]);
     if(hopping_value[index] == SubGhzHopperStateOFF) {
         char text_buf[10] = {0};
-        sprintf(
+        snprintf(
             text_buf,
+            10,
```
Comment (hedger) [inline applications/subghz/scenes/subghz_scene_receiver_config.c]: sizeof()

File: applications/subghz/scenes/subghz_scene_receiver_config.c [hunk 5]
```diff
@@ -160,8 +162,9 @@ void subghz_scene_receiver_config_on_enter(void* context) {
         subghz->scene_manager, SubGhzSceneReceiverConfig, (uint32_t)item);
     variable_item_set_current_value_index(item, value_index);
     char text_buf[10] = {0};
-    sprintf(
+    snprintf(
         text_buf,
+        10,
```
Comment (hedger) [inline applications/subghz/scenes/subghz_scene_receiver_config.c]: sizeof()

**Review by hedger — CHANGES_REQUESTED** [review]

**Review by hedger — APPROVED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**